### PR TITLE
Perception task

### PR DESCRIPTION
**Pull request by mabelzhang**
Just adding the launch file and world file for the perception task, so that @crvogt can get started on the two other tasks.
Added a Tasks page in Wiki: https://github.com/osrf/vorc/wiki/Tasks
Turns out this one wasn't tricky. Am I missing something?

The files are very similar to the VRX ones:
https://github.com/osrf/vrx/blob/master/vrx_gazebo/launch/perception_task.launch
https://github.com/osrf/vrx/blob/master/vrx_gazebo/worlds/perception_task.world.xacro
There are corresponding files for the two other tasks as well (stationkeeping_task and wayfinding_task).

Currently, it's just using the VRX perception task scoring plugin as a tentative example https://github.com/osrf/vrx/blob/master/vrx_gazebo/src/perception_scoring_plugin.cc
We can change it later.

Test with
```
$ roslaunch vorc_gazebo perception_task.launch 
```

@caguero Please check that the behavior is as intended since you are more familiar with what's supposed to happen.

Red & green buoys:
Screenshot from 2020-10-08 04-41-43_crop_scaled

Black buoy:
Screenshot from 2020-10-08 04-41-53_crop_scaled

### CODE REVIEW

**Comment by mabelzhang** [comment]
a9dadc6 and 8d8f713 added basic scoring plugin and perception scoring plugin.
I decided to add these now rather than later, so that we can put topic name `/vorc/task/info` (as opposed to `/vrx/*`) in the Wiki.
Note that they're built with `vorc_` prefix in CMakeLists.txt to differentiate from the VRX ones, so that the SDF loads the right one.

Wiki page on Tasks has been added https://github.com/osrf/vorc/wiki/Tasks

**Review by caguero — COMMENTED** [review]
I wonder if we need to copy back the scoring plugin and the perception scoring plugin. Should we instead parameterize the topics that are hardcoded and replace `/vrx/...` with `/vorc/...`?

**Review by caguero — COMMENTED** [review]
Can we also move the buoys a bit further. They're very close.

File: vorc_gazebo/launch/perception_task.launch [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  <!-- Initial USV location and attitude-->
+  <arg name="x" default="10" />
+  <arg name="y" default="-372" />
+  <arg name="z" default="0.1" />
+  <arg name="P" default="0" />
+  <arg name="R" default="0" />
+  <arg name="Y" default="0.3816" />
+
+  <!-- VORC sensors enabled -->
+  <arg name="vorc_sensors_enabled" default="true" />
```
Comment (caguero) [inline vorc_gazebo/launch/perception_task.launch]: I don't think we're using this parameters as all the sensors are enabled in our case.

File: vorc_gazebo/launch/perception_task.launch [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (caguero) [inline vorc_gazebo/launch/perception_task.launch]: If you look at VRX, there's a `wamv_locked` parameter that is used to lock the boat. With this parameter it shouldn't be possible to move the boat in ` x`, `y`, `z`, and `yaw`. `roll` and `pitch` is free to allow the boat to move with the waves. This feature is not enabled here as I was able to move the boat.

**Review by caguero — CHANGES_REQUESTED** [review]

**Comment by mabelzhang** [comment]
535fc52 should have addressed the inline feedback.
Sensor positions are adjusted with z += 1, as the boat position has been lifted z += 1 in #4 .

> Can we also move the buoys a bit further. They're very close.

How is this distance? I interpreted that you meant move them farther from the boat, or did you mean move the green one farther from the red one?
Screenshot from 2020-10-09 21-22-41_crop

> I wonder if we need to copy back the scoring plugin and the perception scoring plugin.

I copied the scoring plugins thinking we wanted different scoring for VORC. Do we want to just use the same scoring system as VRX? I can parameterize the topic names and delete the new files here if that's what we want.

**Comment by caguero** [comment]
> I copied the scoring plugins thinking we wanted different scoring for VORC. Do we want to just use the same scoring system as VRX? I can parameterize the topic names and delete the new files here if that's what we want.

For now there's no need to have different scoring. If a specific task needs a custom scoring we can create a plugin for it but in this case I think we can reuse it 100%.

**Comment by caguero** [comment]
> [535fc52](https://github.com/osrf/vorc/commit/535fc522cc4d937285134b646888a174c4c776c5) should have addressed the inline feedback.
> Sensor positions are adjusted with z += 1, as the boat position has been lifted z += 1 in #4 .
> 
> > Can we also move the buoys a bit further. They're very close.
> 
> How is this distance? I interpreted that you meant move them farther from the boat, or did you mean move the green one farther from the red one?
> Screenshot from 2020-10-09 21-22-41_crop
> 
> > I wonder if we need to copy back the scoring plugin and the perception scoring plugin.
> 
> I copied the scoring plugins thinking we wanted different scoring for VORC. Do we want to just use the same scoring system as VRX? I can parameterize the topic names and delete the new files here if that's what we want.

**Comment by caguero** [comment]
> How is this distance? I interpreted that you meant move them farther from the boat, or did you mean move the green one farther from the red one?

This is great, thanks!

**Comment by caguero** [comment]
Just waiting for reusing the scoring plugin from VRX.

**Comment by mabelzhang** [comment]
> Just waiting for reusing the scoring plugin from VRX.

eb37c60 removes the C++ files and passes custom topic names to VRX.
I just realized VRX scoring_plugin.cc already looks for custom `task_info_topic` and `contact_debug_topic` parameters in the SDF! Didn't look closely before.

**Review by caguero — APPROVED** [review]
Looks good to me!

**Pull request merged**